server/manager/src/main/java/org/apache/accumulo/manager/TabletGroupWatcher.java

```diff
@@ -876,61 +853,53 @@ private void cancelOfflineTableMigrations(KeyExtent extent) {
     }
   }
 
-  private void repairMetadata(Text row) {
-    Manager.log.debug("Attempting repair on {}", row);
-    // ACCUMULO-2261 if a dying tserver writes a location before its lock information propagates, it
-    // may cause duplicate assignment.
-    // Attempt to find the dead server entry and remove it.
+  private void logIncorrectTabletLocations(TabletMetadata tabletMetadata) {
     try {
       Map<Key,Value> future = new HashMap<>();
       Map<Key,Value> assigned = new HashMap<>();
-      KeyExtent extent = KeyExtent.fromMetaRow(row);
-      String table = AccumuloTable.METADATA.tableName();
-      if (extent.isMeta()) {
-        table = AccumuloTable.ROOT.tableName();
+      KeyExtent extent = tabletMetadata.getExtent();
+      var level = Ample.DataLevel.of(extent.tableId());
+
+      Stream<? extends Entry<Key,Value>> entries;
+      if (level == Ample.DataLevel.ROOT) {
+        RootTabletMetadata rtm = RootTabletMetadata.read(manager.getContext());
+        entries = rtm.getKeyValues();
+      } else {
+        Scanner scanner =
+            manager.getContext().createScanner(level.metaTable(), Authorizations.EMPTY);
+        scanner.fetchColumnFamily(CurrentLocationColumnFamily.NAME);
+        scanner.fetchColumnFamily(FutureLocationColumnFamily.NAME);
+        scanner.setRange(new Range(extent.toMetaRow()));
+        entries = scanner.stream();
       }
-      Scanner scanner = manager.getContext().createScanner(table, Authorizations.EMPTY);
-      scanner.fetchColumnFamily(CurrentLocationColumnFamily.NAME);
-      scanner.fetchColumnFamily(FutureLocationColumnFamily.NAME);
-      scanner.setRange(new Range(row));
-      for (Entry<Key,Value> entry : scanner) {
+
+      entries.forEach(entry -> {
         if (entry.getKey().getColumnFamily().equals(CurrentLocationColumnFamily.NAME)) {
           assigned.put(entry.getKey(), entry.getValue());
         } else if (entry.getKey().getColumnFamily().equals(FutureLocationColumnFamily.NAME)) {
           future.put(entry.getKey(), entry.getValue());
         }
-      }
-      if (!future.isEmpty() && !assigned.isEmpty()) {
-        Manager.log.warn("Found a tablet assigned and hosted, attempting to repair");
-      } else if (future.size() > 1 && assigned.isEmpty()) {
-        Manager.log.warn("Found a tablet assigned to multiple servers, attempting to repair");
-      } else if (future.isEmpty() && assigned.size() > 1) {
-        Manager.log.warn("Found a tablet hosted on multiple servers, attempting to repair");
+      });
+
+      var count = future.size() + assigned.size();
+      if (count <= 1) {
+        Manager.log.info("Tablet {} seems to have correct location based on inspection",
```

Comment:
Wondering if this should be trace level

Reply:
Yeah trace would be good. Was thinking I would like to see if this was getting too many false positives during testing.  However it would probably be best to lower levels to trace during testing and see what is going on in general and if anything jumps out as happening too frequently.